### ISSUE TITLE
Deliver Elements files to Digital Science via S3.

### ISSUE THREAD

**Issue by kelynch** [pulibrary/dspace-development#638]
Per discussion with the vendor to meet ongoing needs of this phase of the migration, deliver files via an S3 bucket to the vendor.

## Acceptance Criteria
- [x] Create a temporary bucket, access policy, and user.
- [x] Securely deliver access credentials to the vendor.
- [x] Confirm access to the bucket with the vendor.
- [x] Deliver files to the S3 bucket. 
- [x] Confirm receipt of files by the vendor.

**Comment by jrgriffiniii** [comment]
Attempting to upload the files to the S3 Bucket using only a browser yields the following for any `princeton.edu` hosts:

![image.png](https://images.zenhubusercontent.com/5b5b6aed981c3204c9ed06c8/d5ad3686-0bd7-457a-a170-0175bac7e8cf)

**Comment by jrgriffiniii** [comment]
Attempting to install the PowerShell tools using the AWS documentation https://docs.aws.amazon.com/powershell/latest/userguide/pstools-s3-upload-object.html yields the following:

![image.png](https://images.zenhubusercontent.com/5b5b6aed981c3204c9ed06c8/02735914-4603-4e87-aeb1-3b1895752112)

**Comment by jrgriffiniii** [comment]
Following the documentation for the `AWSCLI` utilities for Windows, the binary:
![image.png](https://images.zenhubusercontent.com/5b5b6aed981c3204c9ed06c8/32fe5afd-65f1-4751-b2f2-1eb19a4577b9)

**Comment by jrgriffiniii** [comment]
Attempting to interface with the S3 Bucket over the HTTP yields the following error:

![image.png](https://images.zenhubusercontent.com/5b5b6aed981c3204c9ed06c8/f2ae0a92-23ed-4757-997b-c40c2e471ff0)

I am going to attempt to resolve this with a firewall update request on SN@P.

**Comment by jrgriffiniii** [comment]
Requesting to list the contents from the Windows Server environment still seems to be unsuccessful (I receive the same error message). Within the SN@P ticket, I find the following:

> This request record will be closed now that it has been assessed for security concerns, and an individual change record has been submitted for each rule or host addition requested. The rules and host additions have NOT been implemented at this time. Each change request will go through its own approval (if necessary) and fulfillment

I am not certain why requesting traffic over the HTTP for `cisdr201w` is problematic, my explanation was the following:

> Good Afternoon,
>
> This is to please request access for interacting with a dedicated AWS S3 Bucket for transferring files between the cisdr201w.princeton.edu host and Amazon Web Services.
>
> Thank you very much for your assistance and patience.
>
> Sincerely,
> James

**Comment by jrgriffiniii** [comment]
Attempting to connect to this network share from the bastion host (`lib200w`) was not successful:

![image](https://user-images.githubusercontent.com/1443986/171640963-3e4d942b-a2b5-41de-ac2a-8eb1780fcdeb.png)

**Comment by jrgriffiniii** [comment]
I am extremely sorry, this was not the network firewall!

Apparently the Windows Firewall was disabled (along with the policies which were in place). I have just enabled this, and now outbound/inbound traffic behaves as expected:

![image](https://user-images.githubusercontent.com/1443986/171643448-cdd1f2aa-bb3e-42d1-9aee-11fb59d4f5b4.png)

I shall proceed accordingly.

**Comment by kelynch** [comment]
The vendor confirmed the ability to list contents but not download objects.  The IAM policy was adjusted and tested on our end successfully, and this should be resolved now; awaiting confirmation from the vendor.

**Comment by jrgriffiniii** [comment]
The vendor has confirmed that they indeed have access to the S3 Bucket.

**Comment by jrgriffiniii** [comment]
I am having strange behavior regarding trying to checksum the names of files and directories containing whitespace:

```bash
rclone lsf --recursive --fast-list  --ftp-no-check-certificate "symplectic:/princeton_prod_elements/"
Symplectic/
Symplectic/Builds/
Symplectic/Elements/
Symplectic/HR File/
Symplectic/Misc/
Symplectic/Tools/
Symplectic/Tools/MsSqlCmdLnUtils.msi
Symplectic/Tools/msodbcsql.msi
2022/06/06 09:48:40 ERROR : Symplectic/Builds: error listing: directory not found
2022/06/06 09:48:40 ERROR : Symplectic/Elements: error listing: directory not found
Symplectic/HR File/InitialImport.csv
Symplectic/HR File/Princeton_HR_Export_Symplectic_Format.csv
Symplectic/HR File/SymplifyFeed.ps1
Symplectic/HR File/backup/
2022/06/06 09:48:40 ERROR : Symplectic/Misc: error listing: directory not found
2022/06/06 09:48:40 ERROR : Symplectic/HR File/backup: error listing: directory not found
2022/06/06 09:48:40 Failed to lsf with 5 errors: last error was: error in ListJSON: directory not found
```

I have enquired as to whether or not the vendor can access these files.

**Comment by jrgriffiniii** [comment]
I am currently using `rclone` to confirm the integrity of the files within the S3 Bucket (and transfer any problematic/corrupted files):

```bash
Transferred:        1.786 GiB / 13.797 GiB, 13%, 1.567 MiB/s, ETA 2h10m49s
Errors:                 1 (retrying may help)
Checks:             12688 / 22699, 56%
Transferred:         4448 / 14457, 31%
Elapsed time:     17m31.0s
```

**Comment by jrgriffiniii** [comment]
Given the difficulties encountered in attempting to transfer the files between the QA server environment, the FTP server, and the S3 Bucket:

![image.png](https://images.zenhubusercontent.com/5b5b6aed981c3204c9ed06c8/133d3b94-a594-4554-ac13-e07ede4666dc)

...an alternative proposed by @kelynch is to request to proceed with having the new `staging` environment mirror that of the `production` environment (hence, the same files and database exports could be used for both upgraded installations).

**Comment by carolyncole** [comment]
Waiting on signal sciences to verify before we delete the bucket

**Comment by kelynch** [comment]
Digital Science confirmed we can use the production files that we have uploaded for this project both in prod and staging.  They will send us an update on their progress setting up hosted environments on Friday this week.

I am removing "delete the bucket" from this ticket and will set a reminder/make a new ticket for the end of this project as part of cleanup.
